msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLFileWriter.java

```diff
@@ -0,0 +1,562 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+package io.github.msdk.io.mzml2;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.xml.txw2.output.IndentingXMLStreamWriter;
```

Comment:
The `XMLStreamWriter` from javolution and stax do not support indenting of the files, I thought it would be better using this StreamWriter instead for the purpose.